### PR TITLE
Parameterize: fix chlorine transmutation to carbon

### PR DESCRIPTION
**Pull request by raimis**
For a molecule: [3u5j_ligand.mol2.txt](https://github.com/Acellera/htmd/files/1252408/3u5j_ligand.mol2.txt) fails:
```
$ parameterize -m 3u5j_ligand.mol2 -f GAFF2
...
Traceback (most recent call last):
  File "/home/raimis/opt/miniconda3/envs/clean/bin/parameterize", line 11, in <module>
    sys.exit(main_parameterize())
  File "/home/raimis/opt/miniconda3/envs/clean/lib/python3.6/site-packages/htmd/parameterization/cli.py", line 171, in main_parameterize
    outdir=args.outdir)
  File "/home/raimis/opt/miniconda3/envs/clean/lib/python3.6/site-packages/htmd/parameterization/ffmolecule.py", line 101, in __init__
    fftype = FFType(self, method=self.method)
  File "/home/raimis/opt/miniconda3/envs/clean/lib/python3.6/site-packages/htmd/parameterization/fftype.py", line 102, in __init__
    os.path.join(tmpdir, 'mol.frcmod'))
  File "/home/raimis/opt/miniconda3/envs/clean/lib/python3.6/site-packages/htmd/parameterization/ff.py", line 714, in __init__
    idx = self.index_by_name[ff[1]]
KeyError: 'CL'
```
Because chlorine is changed to carbon.

Depends on #397.

### CODE REVIEW

**Comment by raimis** [comment]
@j3mdamas can you review this, please!

**Comment by j3mdamas** [comment]
After today's release

**Comment by raimis** [comment]
Ping @j3mdamas

**Comment by raimis** [comment]
@stefdoerr can you review this, please.

File: htmd/parameterization/ffmolecule.py [hunk 1]
```diff
@@ -126,33 +126,30 @@ def _rename_mol(self):
         # This fixes up the atom naming and reside name to be consistent
         # NB this scheme matches what MATCH does. Don't change it
         # Or the naming will be inconsistent with the RTF
-        import re
 
-        hh = dict()
+        sufices = dict()
 
+        print('\nRename atoms:')
```
Comment (stefdoerr) [inline htmd/parameterization/ffmolecule.py]: Any specific reason why you use print instead of a logger as we do in the rest of HTMD?

Comment (raimis) [reply htmd/parameterization/ffmolecule.py]: So far, loggers nowhere to be seen in `parameterize`...

Comment (raimis) [reply htmd/parameterization/ffmolecule.py]: This is just a bug fix not a rewriting.

Comment (stefdoerr) [reply htmd/parameterization/ffmolecule.py]: Ok, it was just a general consideration. Maybe we should make an issue on it to move prints to loggers once the main refactoring is done

File: htmd/parameterization/ffmolecule.py [hunk 2]
```diff
@@ -126,33 +126,30 @@ def _rename_mol(self):
...
-            # Remove any character that isn't alpha
-            t = re.sub('[^A-Z]*', "", t )
-            if t != self.name[i].upper():
-#               print(" Atom #%d renamed from [%s] to [%s]" %(i, self.name[i], t ) )
-               pass
+            name = self.name[i].upper()
+
+            # This fixes the specific case where a name is 3 or 4 characters, as X-TOOL seems to make
+            if re.match('^[A-Z]{3,4}$', name):
+               name = name[:-2] # Remove the last 2 characters
```
Comment (stefdoerr) [inline htmd/parameterization/ffmolecule.py]: Last question. Sorry for my lack of knowledge of the specific problem here, but do you really want to remove the last 2 characters? So if it's 4 characters you keep 2 and if it's 3 characters you keep 1? Or you want just the first character?

Comment (raimis) [reply htmd/parameterization/ffmolecule.py]: Yes, this is a convention of X-Tool, where the atom name is: element symbol + 2 character. So I want to get rid of these 2 characters. :)

Comment (stefdoerr) [reply htmd/parameterization/ffmolecule.py]: ok fine

**Pull request merged**